spec/unit/kickstart_spec.rb

```diff
@@ -0,0 +1,195 @@
+          status: "changed",
+          changes: ["link_path"],
+          mode: "777",
+          user: "root",
+          group: "root",
+          type: "link",
+          target: "/opt/test-quote-char/target-with-quote'-foo"
+        )
+      kickstart = Machinery::Kickstart.new(description)
+      @output_dir = given_directory
+      kickstart.profile(@output_dir)
+      generated_profile = File.read(File.join(@output_dir,"ks.cfg"))
+      expect(generated_profile).to include(
+        "ln -s '/opt/test-quote-char/target-with-quote'\\\''-foo' '/mnt/sysimage/opt/test-quote-char/link'"
+      )
+    end
+
+    it "creates the expected profile" do
+      kickstart = Machinery::Kickstart.new(description)
+      @output_dir = given_directory
+      kickstart.profile(@output_dir)
+      generated_profile = File.read(File.join(@output_dir,"ks.cfg"))
+      expect(generated_profile).to eq(expected_profile)
+    end
+
+    it "does not ask for export URL if files weren't extracted" do
+      [
+        "changed_config_files",
+        "changed_managed_files",
+        "unmanaged_files"
+      ].each do |scope|
+        description[scope].extracted = false
+      end
+      kickstart = Machinery::Kickstart.new(description)
+      @output_dir = given_directory
+      kickstart.profile(@output_dir)
+      generated_profile = File.read(File.join(@output_dir,"ks.cfg"))
+      expect(generated_profile).not_to include("Enter URL to system description")
+    end
+  end
+
+  describe "#write" do
+    let(:ip) { "192.168.0.35" }
+    before(:each) do
+      kickstart = Machinery::Kickstart.new(description)
+      @output_dir = given_directory
+      allow(kickstart).to receive(:outgoing_ip).and_return(ip)
+      kickstart.write(@output_dir)
+      expect(captured_machinery_output).to include(
+        "Note: The permssions of the KickStart directory are restricted to be only" \
+          " accessible by the current user. Further instructions are provided by the " \
+          "README.md in the exported directory."
+      )
+    end
+
+    it "copies over the system description" do
+      expect(File.exist?(File.join(@output_dir, "manifest.json"))).to be(true)
+    end
+
+    it "adds the ks.cfg" do
+	    expect(File.exist?(File.join(@output_dir, "ks.cfg"))).to be(true)
+    end
+
+    it "adds unmanaged files filter list" do
+      expect(File.exist?(File.join(@output_dir, "unmanaged_files_kickstart_excludes"))).to be(true)
+    end
+
+    it "filters log files from the Kickstart export" do
+      expect(File.read(File.join(@output_dir, "unmanaged_files_kickstart_excludes"))).
+        to include("var/log/*")
+    end
+
+    it "adds the kickstart export readme" do
+      expect(File.exist?(File.join(@output_dir, "README.md"))).to be(true)
+    end
+
+    it "adds the ip of the outgoing network to the readme" do
+      file = File.read(File.join(@output_dir, "README.md"))
+      expect(file).to include("ks=http://#{ip}:8000/ks.cfg")      
```

Comment:
I know that we also do this in the Autoyast test, but I think it is better to expect the whole string for readability and to prevent issues if the value is empty by accident:
```
expect(file).to include("ks=http://192.168.0.35:8000/ks.cfg")
```